azuki-script-generation/src/main/kotlin/com/anaplan/engineering/azuki/script/generation/ScriptGenerator.kt

```diff
@@ -61,32 +57,36 @@ abstract class ScriptGenerator<
         })
 
     fun generateThenScriptFromChecks(checks: List<Check>): String {
-        val body = ScriptGenerationCheckStateBuilder(checkStateFactory).build(checks)
+        checks.forEach {
+            require(it !is UnsupportedCheck) { "unsupported check: $it" }
+            require(it is ScriptGenerationCheck<*>) { "check $it is not a ScriptGenerationCheck" }
+        }
+        // compose everything maximally before we start trying to resolve the checks
+        val resolvers = checks.filterIsInstance<ScriptGenerationCheck<E>>().map { it.composeInto(environment) }
```

Comment:
The `filterIsInstance` here is an awkward way of forcing an `E`.  I guess I could push it into the `map`.